### ISSUE TITLE
[Bug] Assertion failed: rotary_emb.shape[0] * rotary_emb.shape[1] == M when setting 2 in the batch_size of the empty latent image

### ISSUE THREAD

**Issue by ryukbk** [nunchux-ai/nunchaku#932]
### Checklist

- [x] 1. I have searched for related issues and FAQs (https://nunchaku.tech/docs/nunchaku/faq/faq.html) but was unable to find a solution.
- [x] 2. The issue persists in the latest version.
- [x] 3. Please note that without environment information and a minimal reproducible example, it will be difficult for us to reproduce and address the issue, which may delay our response.
- [x] 4. If your report is a question rather than a bug, please submit it as a discussion at https://github.com/mit-han-lab/nunchaku/discussions/new/choose. Otherwise, this issue will be closed.
- [x] 5. If this is related to ComfyUI, please report it at https://github.com/mit-han-lab/ComfyUI-nunchaku/issues.
- [x] 6. I will do my best to describe the issue in English.

### Describe the Bug

When setting 2 in the batch_size of the empty latent image for the latest version of nunchaku and Z-image, I get this Assertion error which I don't get when I use Z-image withouth nunchaku:

`Assertion failed: rotary_emb.shape[0] * rotary_emb.shape[1] == M, file C:\Users\nunchaku\Desktop\actions-runner\_work\nunchaku\nunchaku\src\kernels\zgemm\gemm_w4a4_launch_impl.cuh, line 353
Fatal Python error: Aborted

Stack (most recent call first):
  File "J:\ai\ComfyUI_windows_portable\python_embeded\Lib\site-packages\nunchaku\ops\gemm.py", line 130 in svdq_gemm_w4a4_cuda
  File "J:\ai\ComfyUI\custom_nodes\ComfyUI-nunchaku\models\zimage.py", line 107 in fused_qkv_norm_rotary
  File "J:\ai\ComfyUI\custom_nodes\ComfyUI-nunchaku\models\zimage.py", line 162 in forward
  File "J:\ai\ComfyUI_windows_portable\python_embeded\Lib\site-packages\torch\nn\modules\module.py", line 1832 in inner
  File "J:\ai\ComfyUI_windows_portable\python_embeded\Lib\site-packages\torch\nn\modules\module.py", line 1884 in _call_impl
  File "J:\ai\ComfyUI_windows_portable\python_embeded\Lib\site-packages\torch\nn\modules\module.py", line 1778 in _wrapped_call_impl
  File "J:\ai\ComfyUI_windows_portable\ComfyUI\comfy\ldm\lumina\model.py", line 340 in forward
  File "J:\ai\ComfyUI_windows_portable\python_embeded\Lib\site-packages\torch\nn\modules\module.py", line 1789 in _call_impl
  File "J:\ai\ComfyUI_windows_portable\python_embeded\Lib\site-packages\torch\nn\modules\module.py", line 1778 in _wrapped_call_impl
  File "J:\ai\ComfyUI_windows_portable\ComfyUI\comfy\ldm\lumina\model.py", line 761 in patchify_and_embed
  File "J:\ai\ComfyUI_windows_portable\ComfyUI\comfy\ldm\lumina\model.py", line 841 in _forward
  File "J:\ai\ComfyUI_windows_portable\ComfyUI\comfy\patcher_extension.py", line 112 in execute
  File "J:\ai\ComfyUI_windows_portable\ComfyUI\comfy\ldm\lumina\model.py", line 804 in forward
  File "J:\ai\ComfyUI_windows_portable\python_embeded\Lib\site-packages\torch\nn\modules\module.py", line 1832 in inner
  File "J:\ai\ComfyUI_windows_portable\python_embeded\Lib\site-packages\torch\nn\modules\module.py", line 1884 in _call_impl
  File "J:\ai\ComfyUI_windows_portable\python_embeded\Lib\site-packages\torch\nn\modules\module.py", line 1778 in _wrapped_call_impl
  File "J:\ai\ComfyUI_windows_portable\ComfyUI\comfy\model_base.py", line 217 in _apply_model
  File "J:\ai\ComfyUI_windows_portable\ComfyUI\comfy\patcher_extension.py", line 112 in execute
  File "J:\ai\ComfyUI_windows_portable\ComfyUI\comfy\model_base.py", line 178 in apply_model
  File "J:\ai\ComfyUI_windows_portable\ComfyUI\comfy\samplers.py", line 325 in _calc_cond_batch
  File "J:\ai\ComfyUI_windows_portable\ComfyUI\comfy\patcher_extension.py", line 112 in execute
  File "J:\ai\ComfyUI_windows_portable\ComfyUI\comfy\samplers.py", line 213 in _calc_cond_batch_outer
  File "J:\ai\ComfyUI_windows_portable\ComfyUI\comfy\samplers.py", line 205 in calc_cond_batch
  File "J:\ai\ComfyUI_windows_portable\ComfyUI\comfy\samplers.py", line 380 in sampling_function
  File "J:\ai\ComfyUI_windows_portable\ComfyUI\comfy\samplers.py", line 964 in predict_noise
  File "J:\ai\ComfyUI_windows_portable\ComfyUI\comfy\patcher_extension.py", line 112 in execute
  File "J:\ai\ComfyUI_windows_portable\ComfyUI\comfy\samplers.py", line 961 in outer_predict_noise
  File "J:\ai\ComfyUI_windows_portable\ComfyUI\comfy\samplers.py", line 954 in __call__
  File "J:\ai\ComfyUI_windows_portable\ComfyUI\comfy\samplers.py", line 400 in __call__
  File "J:\ai\ComfyUI_windows_portable\ComfyUI\comfy\k_diffusion\sampling.py", line 752 in sample_dpmpp_sde
  File "J:\ai\ComfyUI_windows_portable\python_embeded\Lib\site-packages\torch\utils\_contextlib.py", line 124 in decorate_context
  File "J:\ai\ComfyUI_windows_portable\ComfyUI\comfy\k_diffusion\sampling.py", line 975 in sample_dpmpp_sde_gpu
  File "J:\ai\ComfyUI_windows_portable\python_embeded\Lib\site-packages\torch\utils\_contextlib.py", line 124 in decorate_context
  File "J:\ai\ComfyUI_windows_portable\ComfyUI\comfy\samplers.py", line 751 in sample
  File "J:\ai\ComfyUI_windows_portable\ComfyUI\comfy\patcher_extension.py", line 112 in execute
  File "J:\ai\ComfyUI_windows_portable\ComfyUI\comfy\samplers.py", line 981 in inner_sample
  File "J:\ai\ComfyUI_windows_portable\ComfyUI\comfy\samplers.py", line 995 in outer_sample
  File "J:\ai\ComfyUI_windows_portable\ComfyUI\comfy\patcher_extension.py", line 112 in execute
  File "J:\ai\ComfyUI_windows_portable\ComfyUI\comfy\samplers.py", line 1052 in sample
  File "J:\ai\ComfyUI_windows_portable\ComfyUI\comfy\samplers.py", line 1070 in sample
  File "J:\ai\ComfyUI_windows_portable\ComfyUI\comfy\samplers.py", line 1180 in sample
  File "J:\ai\ComfyUI_windows_portable\ComfyUI\comfy\sample.py", line 66 in sample
  File "J:\ai\ComfyUI_windows_portable\ComfyUI\nodes.py", line 1537 in common_ksampler
  File "J:\ai\ComfyUI_windows_portable\ComfyUI\nodes.py", line 1572 in sample
  File "J:\ai\ComfyUI_windows_portable\ComfyUI\execution.py", line 296 in process_inputs
  File "J:\ai\ComfyUI_windows_portable\ComfyUI\execution.py", line 308 in _async_map_node_over_list
  File "J:\ai\ComfyUI_windows_portable\ComfyUI\execution.py", line 334 in get_output_data
  File "J:\ai\ComfyUI_windows_portable\ComfyUI\execution.py", line 534 in execute
  File "J:\ai\ComfyUI_windows_portable\ComfyUI\execution.py", line 770 in execute_async
  File "asyncio\events.py", line 89 in _run
  File "asyncio\base_events.py", line 2050 in _run_once
  File "asyncio\base_events.py", line 683 in run_forever
  File "asyncio\base_events.py", line 712 in run_until_complete
  File "asyncio\runners.py", line 118 in run
  File "asyncio\runners.py", line 195 in run
  File "J:\ai\ComfyUI_windows_portable\ComfyUI\execution.py", line 711 in execute
  File "J:\ai\ComfyUI_windows_portable\ComfyUI\main.py", line 313 in prompt_worker
  File "threading.py", line 994 in run
  File "threading.py", line 1043 in _bootstrap_inner
  File "threading.py", line 1014 in _bootstrap

Extension modules: sqlalchemy.cyextension.collections, sqlalchemy.cyextension.immutabledict, sqlalchemy.cyextension.processors, sqlalchemy.cyextension.resultproxy, sqlalchemy.cyextension.util, greenlet._greenlet, markupsafe._speedups, yaml._yaml, PIL._imaging, multidict._multidict, yarl._quoting_c, propcache._helpers_c, _brotli, aiohttp._http_writer, aiohttp._http_parser, aiohttp._websocket.mask, aiohttp._websocket.reader_c, frozenlist._frozenlist, charset_normalizer.md, numpy._core._multiarray_umath, numpy.linalg._umath_linalg, torch._C, torch._C._dynamo.autograd_compiler, torch._C._dynamo.eval_frame, torch._C._dynamo.guards, torch._C._dynamo.utils, torch._C._fft, torch._C._linalg, torch._C._nested, torch._C._nn, torch._C._sparse, torch._C._special, cuda.bindings._bindings.cydriver, cuda.bindings.cydriver, cuda.bindings.driver, cuda.bindings._bindings.cyruntime_ptds, cuda.bindings._bindings.cyruntime, cuda.bindings.cyruntime, cuda.bindings.runtime, numpy.random._common, numpy.random.bit_generator, numpy.random._bounded_integers, numpy.random._pcg64, numpy.random._mt19937, numpy.random._generator, numpy.random._philox, numpy.random._sfc64, numpy.random.mtrand, psutil._psutil_windows, PIL._imagingft, av._core, av.logging, av.bytesource, av.buffer, av.audio.format, av.error, av.dictionary, av.container.pyio, av.option, av.descriptor, av.format, av.utils, av.stream, av.container.streams, av.sidedata.encparams, av.sidedata.motionvectors, av.sidedata.sidedata, av.opaque, av.packet, av.container.input, av.container.output, av.container.core, av.codec.context, av.video.format, av.video.reformatter, av.plane, av.video.plane, av.video.frame, av.video.stream, av.codec.hwaccel, av.codec.codec, av.frame, av.audio.layout, av.audio.plane, av.audio.frame, av.audio.stream, av.filter.link, av.filter.context, av.filter.graph, av.filter.filter, av.filter.loudnorm, av.audio.resampler, av.audio.codeccontext, av.audio.fifo, av.bitstream, av.video.codeccontext, cython.cimports.libc.math, _cyutility, scipy._cyutility, scipy._lib._ccallback_c, scipy.ndimage._nd_image, scipy.ndimage._rank_filter_1d, scipy.special._ufuncs_cxx, scipy.special._ellip_harm_2, scipy.special._special_ufuncs, scipy.special._gufuncs, scipy.special._ufuncs, scipy.special._specfun, scipy.special._comb, _ni_label, scipy.ndimage._ni_label, regex._regex, scipy.integrate._odepack, scipy.integrate._quadpack, scipy.integrate._vode, scipy.integrate._dop, scipy.sparse._sparsetools, _csparsetools, scipy.sparse._csparsetools, scipy.linalg._fblas, scipy.linalg._flapack, scipy.linalg.cython_lapack, scipy.linalg._cythonized_array_utils, scipy.linalg._solve_toeplitz, scipy.linalg._batched_linalg, scipy.linalg._decomp_lu_cython, scipy.linalg._matfuncs_schur_sqrtm, scipy.linalg._matfuncs_expm, scipy.linalg._linalg_pythran, scipy.linalg.cython_blas, scipy.linalg._decomp_update, scipy.sparse.linalg._dsolve._superlu, scipy.sparse.linalg._eigen.arpack._arpacklib, scipy.sparse.linalg._propack, scipy.optimize._group_columns, scipy._lib.messagestream, scipy.optimize._trlib._trlib, scipy.optimize._lbfgsb, _moduleTNC, scipy.optimize._moduleTNC, scipy.optimize._slsqplib, scipy.optimize._minpack, scipy.optimize._lsq.givens_elimination, scipy.optimize._zeros, scipy._lib._uarray._uarray, scipy.linalg._decomp_interpolative, scipy.optimize._bglu_dense, scipy.optimize._lsap, scipy.spatial._ckdtree, scipy.spatial._qhull, scipy.spatial._voronoi, scipy.spatial._hausdorff, scipy.spatial._distance_wrap, scipy.spatial.transform._rotation_cy, scipy.spatial.transform._rigid_transform_cy, scipy.optimize._direct, scipy.interpolate._fitpack, scipy.interpolate._dfitpack, scipy.interpolate._dierckx, scipy.interpolate._ppoly, scipy.interpolate._interpnd, scipy.interpolate._rbfinterp_pythran, scipy.interpolate._rgi_cython, scipy.special.cython_special, scipy.stats._stats, scipy.stats._biasedurn, scipy.stats._stats_pythran, scipy.stats._levy_stable.levyst, scipy.stats._ansari_swilk_statistics, scipy.sparse.csgraph._tools, scipy.sparse.csgraph._shortest_path, scipy.sparse.csgraph._traversal, scipy.sparse.csgraph._min_spanning_tree, scipy.sparse.csgraph._flow, scipy.sparse.csgraph._matching, scipy.sparse.csgraph._reordering, scipy.stats._sobol, scipy.stats._qmc_cy, scipy.stats._rcont.rcont, scipy.stats._qmvnt_cy, av.subtitles.stream, scipy.io.matlab._mio_utils, scipy.io.matlab._streams, scipy.io.matlab._mio5_utils, kiwisolver._cext, skimage.measure._ccomp, skimage.measure._moments_cy, skimage.measure._find_contours_cy, skimage.measure._marching_cubes_lewiner_cy, _cffi_backend, sklearn.__check_build._check_build, pandas._libs._cyutility, pandas._libs.tslibs.ccalendar, pandas._libs.tslibs.np_datetime, pandas._libs.tslibs.dtypes, pandas._libs.tslibs.base, pandas._libs.tslibs.nattype, pandas._libs.tslibs.timezones, pandas._libs.properties, pandas._libs.tslibs.fields, pandas._libs.tslibs.timedeltas, pandas._libs.tslibs.tzconversion, pandas._libs.tslibs.timestamps, pandas._libs.tslibs.offsets, pandas._libs.tslibs.strptime, pandas._libs.tslibs.parsing, pandas._libs.tslibs.conversion, pandas._libs.tslibs.period, pandas._libs.tslibs.vectorized, pandas._libs.ops_dispatch, pandas._libs.missing, pandas._libs.hashtable, pandas._libs.algos, pandas._libs.interval, pandas._libs.lib, pandas._libs.ops, pandas._libs.hashing, pandas._libs.arrays, pandas._libs.tslib, pandas._libs.sparse, pandas._libs.internals, pandas._libs.indexing, pandas._libs.index, pandas._libs.writers, pandas._libs.join, pandas._libs.window.aggregations, pandas._libs.window.indexers, pandas._libs.reshape, pandas._libs.groupby, pandas._libs.json, pandas._libs.parsers, pandas._libs.testing, sklearn._cyutility, sklearn.utils._isfinite, sklearn.utils.sparsefuncs_fast, sklearn.utils.murmurhash, sklearn.utils._openmp_helpers, sklearn.metrics.cluster._expected_mutual_info_fast, sklearn.metrics._dist_metrics, sklearn.metrics._pairwise_distances_reduction._datasets_pair, sklearn.utils._cython_blas, sklearn.metrics._pairwise_distances_reduction._base, sklearn.metrics._pairwise_distances_reduction._middle_term_computer, sklearn.utils._heap, sklearn.utils._sorting, sklearn.metrics._pairwise_distances_reduction._argkmin, sklearn.metrics._pairwise_distances_reduction._argkmin_classmode, sklearn.utils._vector_sentinel, sklearn.metrics._pairwise_distances_reduction._radius_neighbors, sklearn.metrics._pairwise_distances_reduction._radius_neighbors_classmode, sklearn.metrics._pairwise_fast, sklearn.preprocessing._csr_polynomial_expansion, sklearn.preprocessing._target_encoder_fast, sentencepiece._sentencepiece, PIL._webp, numba.core.typeconv._typeconv, numba._helperlib, numba._dynfunc, numba._dispatcher, numba.core.runtime._nrt_python, _win32sysloader, win32api, numba.np.ufunc._internal, numba.experimental.jitclass._box, websockets.speedups, google._upb._message, simsimd, stringzilla, insightface.thirdparty.face3d.mesh.cython.mesh_core_cython, PIL._imagingcms, msgpack._cmsgpack, scipy.signal._sigtools, scipy.signal._max_len_seq_inner, scipy.signal._upfirdn_apply, scipy.signal._spline, scipy.signal._sosfilt, scipy.signal._peak_finding_utils, lxml._elementpath, lxml.etree, pycocotools._mask, skimage.morphology._misc_cy, _skeletonize_lee_cy, skimage.morphology._skeletonize_lee_cy, skimage.morphology._skeletonize_various_cy, skimage._shared.geometry, skimage.measure._pnpoly, skimage.morphology._convex_hull, skimage.morphology._grayreconstruct, skimage.morphology._extrema_cy, skimage.morphology._flood_fill_cy, skimage.morphology._max_tree, xxhash._xxhash, pywt._extensions._dwt, pywt._extensions._cwt, pywt._extensions._pywt, pywt._extensions._swt, skimage.transform._warps_cy, skimage.feature._canny_cy (total: 298)`

### Environment

Windows 11, PyTorch 2.12, CUDA 13.2

### Reproduction Steps

Set 2 in the batch size of the empty latent image and generate an image with Z-image
